File: roller_flutter_app/lib/src/core/auth/staff_chat_access.dart
/// Quién puede ver y escribir en el chat staff del grupo.
bool canAccessStaffChat(Map<String, dynamic>? me) {
  if (me == null) return false;
  final tipo = (me['tipoPerfil'] ?? '').toString();
  if (tipo == 'administrador' || tipo == 'liderGrupo') return true;
  final grupoId = (me['grupoId'] ?? me['grupo_id'] ?? '').toString().trim();
  return grupoId.isNotEmpty;
}

File: SIIG-ROLLER-BACKEND/src/utils/staffChatAccess.js
/**
 * Chat staff: administradores, líderes y miembros con grupo asignado.
 */
function canAccessStaffChat(user) {
  if (!user) return false;
  const tipo = user.tipoPerfil || user.tipo_perfil;
  if (tipo === 'administrador' || tipo === 'liderGrupo') return true;
  const grupoId = user.grupoId ?? user.grupo_id;
  return grupoId != null && String(grupoId).trim() !== '';
}

module.exports = {canAccessStaffChat};
